### PR TITLE
mount + pipeline

### PR DESCRIPTION
**Pull request by mrkurt**
These are functions we've been using all over that are stable enough to include in the core now. A `mount` function for routing path prefixes, and a `pipeline` for composing fetch generators.

### CODE REVIEW

File: v8env/test/fetch.pipeline.spec.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { expect } from 'chai'
+
+import pipeline from '@fly/fetch/pipeline'
```
Comment (michaeldwan) [inline v8env/test/fetch.pipeline.spec.js]: You'll need to add a glob here I think https://github.com/superfly/fly/pull/130/files#diff-9fe450fad459952e98a9293481a9c819R110
```
ERROR in ./v8env/test/fetch.pipeline.spec.js
Module not found: Error: Can't resolve '@fly/fetch/pipeline' in '/Users/md/src/superfly/fly/v8env/test'
resolve '@fly/fetch/pipeline' in '/Users/md/src/superfly/fly/v8env/test'
  Parsed request is a module
  using description file: /Users/md/src/superfly/fly/v8env/package.json (relative path: ./test)
    Field 'browser' doesn't contain a valid alias configuration
  after using description file: /Users/md/src/superfly/fly/v8env/package.json (relative path: ./test)
    resolve as module
      /Users/md/src/superfly/fly/v8env/test/node_modules doesn't exist or is not a directory
      /Users/md/src/superfly/node_modules doesn't exist or is not a directory
      /Users/md/src/node_modules doesn't exist or is not a directory
      /Users/md/node_modules doesn't exist or is not a directory
      /Users/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
      looking for modules in /Users/md/src/superfly/fly/v8env/node_modules
        using description file: /Users/md/src/superfly/fly/v8env/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /Users/md/src/superfly/fly/v8env/package.json (relative path: ./node_modules)
          using description file: /Users/md/src/superfly/fly/v8env/package.json (relative path: ./node_modules/@fly/fetch/pipeline)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/md/src/superfly/fly/v8env/node_modules/@fly/fetch/pipeline doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/md/src/superfly/fly/v8env/node_modules/@fly/fetch/pipeline.js doesn't exist
            .png
              Field 'browser' doesn't contain a valid alias configuration
              /Users/md/src/superfly/fly/v8env/node_modules/@fly/fetch/pipeline.png doesn't exist
            as directory
              /Users/md/src/superfly/fly/v8env/node_modules/@fly/fetch/pipeline doesn't exist
      looking for modules in /Users/md/src/superfly/fly/node_modules
        using description file: /Users/md/src/superfly/fly/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /Users/md/src/superfly/fly/package.json (relative path: ./node_modules)
          using description file: /Users/md/src/superfly/fly/package.json (relative path: ./node_modules/@fly/fetch/pipeline)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/md/src/superfly/fly/node_modules/@fly/fetch/pipeline doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/md/src/superfly/fly/node_modules/@fly/fetch/pipeline.js doesn't exist
            .png
              Field 'browser' doesn't contain a valid alias configuration
              /Users/md/src/superfly/fly/node_modules/@fly/fetch/pipeline.png doesn't exist
            as directory
              /Users/md/src/superfly/fly/node_modules/@fly/fetch/pipeline doesn't exist
```

File: v8env/src/fly/fetch/proxy.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  if (origin.pathname && origin.pathname.length > 0) {
+    url.pathname = [origin.pathname.replace(/\/$/, ''), url.pathname.replace(/^\//, "")].join("/")
+  }
+  if (url.pathname.startsWith("//")) {
+    url.pathname = url.pathname.substring(1)
+  }
+
+  breq.url = url.toString()
+  // we extend req with remoteAddr
+  breq.headers.set("x-forwarded-for", (<any>req).remoteAddr)
```
Comment (michaeldwan) [inline v8env/src/fly/fetch/proxy.ts]: Don't need the any cast anymore 😄 https://github.com/superfly/fly/blob/master/v8env/src/request.ts#L31

File: v8env/src/fly/fetch/proxy.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+/**
+ * @protected
+ * @hidden
+ * @param origin 
+ * @param options 
+ * @param req 
+ * @param init 
+ */
+export function buildProxyRequest(origin: string | URL, options: ProxyOptions, req: RequestInfo, init?: RequestInit) {
```
Comment (michaeldwan) [inline v8env/src/fly/fetch/proxy.ts]: does this need to be exported?

Comment (mrkurt) [reply v8env/src/fly/fetch/proxy.ts]: It does for the tests!

File: v8env/test/fetch.pipeline.spec.js (hunk 1 above)
Comment (mrkurt) [reply v8env/test/fetch.pipeline.spec.js]: That shouldn't need a glob, but you will need to run `build:core` to get it to work.

Comment (michaeldwan) [reply v8env/test/fetch.pipeline.spec.js]: Whoops, it's on my end... removed dist & lib and it's fine now 😭

**Pull request merged**